src/LoggingFormats.jl

```diff
@@ -1,5 +1,40 @@
 module LoggingFormats
 
-greet() = print("Hello World!")
+export Truncated
+
+using Logging
+
+function shorten_str(str, max_len)
+    suffix = "…"
+    if length(str) > max_len
+        str[1:nextind(str, 0, min(end, max_len-length(suffix)))] * suffix
+    else
+        str
+    end
+end
```

Comment:
```suggestion
shorten_str(str, max_len) = shorten_str(string(str), max_len)
function shorten_str(str::String, max_len)
    if textwidth(str) <= max_len
        return SubString(str)
    end
    len = textwidth('…')
    ind = 1
    for i in eachindex(str)
        c = @inbounds str[i]
        len += textwidth(c)
        len > max_len && break
        ind = i
    end
    return SubString(str, 1, ind)
end
```

Reply:
That looks good, but I would like to keep the '…' at the end of string to show it's been cut.